File: batch/amendements/getAmdmtsOrder.pl
#!/usr/bin/perl

use WWW::Mechanize;
use HTML::TokeParser;
$count = 0;

$loi = shift;

$url = "http://recherche2.assemblee-nationale.fr/amendements/resultats.jsp?typeEcran=avance&chercherDateParNumero=non&NUM_INIT=".$loi."&NUM_AMEND=&AUTEUR=&DESIGNATION_ARTICLE=&DESIGNATION_ALINEA=&SORT_EN_SEANCE=&DELIBERATION=&NUM_PARTIE=&DateDebut=&DateFin=&periode=&LEGISLATURE=13Amendements&QueryText=&Scope=TEXTEINTEGRAL&SortField=ORDRE_TEXTE&SortOrder=Asc&format=HTML&ResultCount=5000&searchadvanced=Rechercher";
$a = WWW::Mechanize->new();
$a->get($url);
$content = $a->content;
$p = HTML::TokeParser->new(\$content);
open FILE, ">:utf8", "liasse_order.tmp";
	
while ($t = $p->get_tag('a')) {
    if ($t->[1]{class} eq 'lienamendement') {
	$htmfile = $t->[1]{href};
	next if ($htmfile =~ /(index|javascript)/);
	if ($htmfile =~ /\d{4}.(\d{4})\.asp/) {
	    $num = $1;
	    $num =~ s/^0*//;
	    print FILE $num."\n";
	    $count++;
	}
    }
}
close FILE;
print $count." amendements pour la loi ".$loi."\n";

File: batch/amendements/getAmdmtsOrder.py
#!/usr/bin/env python

import requests, json, re, sys

if len(sys.argv) != 3:
    print "USAGE: python getAmdmtsOrder.py <ID_DOSSIER> <ID_EXAMEN>"
    exit(1)

id_dossier = sys.argv[1]
id_examen = sys.argv[2]

url = "http://www2.assemblee-nationale.fr/recherche/query_amendements?typeDocument=amendement&idExamen=%s&idDossierLegislatif=%s&numAmend=&idAuteur=&idArticle=&idAlinea=&sort=&dateDebut=&dateFin=&periodeParlementaire=&texteRecherche=&rows=2500&format=html&tri=ordreTexteasc&typeRes=liste&start=" % (id_examen, id_dossier)

re_clean_num = re.compile(r"\D")
count = 0
with open ("liasse_order.tmp", "w") as f:
    for line in requests.get(url).json()["data_table"]:
        print >> f, re_clean_num.sub("", line.split("|")[5])
        count += 1

print "%s amendements" % count

